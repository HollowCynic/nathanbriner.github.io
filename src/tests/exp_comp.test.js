import { render, screen } from '@testing-library/react';
import {MemoryRouter} from 'react-router-dom'
import Component1 from '../components/experiment_components/exp_comp1'
import Component1b from '../components/experiment_components/exp_comp1b'
import Component3 from '../components/experiment_components/exp_comp3'
import Trail from '../components/experiment_components/exp_comp4_spring'


test('test experiment comp 1', () => {
    render(<Component1 />, {wrapper:MemoryRouter});
    const linkElement = screen.getByText(/Component 1/i);
    expect(linkElement).toBeInTheDocument();
  });

test('test experiment comp 1b', () => {
    render(<Component1b />, {wrapper:MemoryRouter});
    const linkElement2 = screen.getByText(/component 1b/i);
    expect(linkElement2).toBeInTheDocument();
});

test('test experiment comp 1b in a different way', () => {
    const {asFragment} = render(<Component1b/>)
    render(<Component1b />, {wrapper:MemoryRouter});
    expect(asFragment(<Component1b/>)).toMatchSnapshot() 
});

test('test experiment comp 3', () => {
    const {asFragment} = render(<Component3/>)
    render(<Component3 />, {wrapper:MemoryRouter});
    expect(asFragment(<Component1b/>)).toMatchSnapshot() 
});

test('test experiment comp 4, Trail', () => {
    render(<Trail />, {wrapper:MemoryRouter});
    const linkElement = screen.getByText(/usetrail/i);
    expect(linkElement).toBeInTheDocument();
  });


