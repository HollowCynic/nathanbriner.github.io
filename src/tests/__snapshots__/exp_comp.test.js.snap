// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test experiment comp 1b in a different way 1`] = `
<DocumentFragment>
  <div
    style="margin-top: -500px;"
  >
    <div
      style="background: slateblue; color: white; padding: 1.5rem;"
    >
      <h1>
        Component 1B
      </h1>
      <p>
        This object should drop down from top of the screen and fade in
      </p>
      <button
        class="button-class"
      >
        This Button Does Nothing
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`test experiment comp 3 1`] = `
<DocumentFragment>
  <div
    style="opacity: 1;"
  >
    <div
      style="background: skyblue; color: white; padding: 1.5rem; margin-bottom: 0px;"
    >
      <h1>
        Component 3
      </h1>
      <p
        style="color: grey; font-size: 18px;"
      >
        This is the third component. 
        <b>
          Click
        </b>
         the component to fade out
      </p>
    </div>
  </div>
</DocumentFragment>
`;
